src/py/flwr/server/compat/driver_client_proxy_test.py

```diff
@@ -17,348 +17,277 @@
-        with self.assertRaises(ValueError):
-            validate_task_res(task_res=task_res)
+        # Execute and assert
+        self.assertRaises(
+            Exception, self.client.get_parameters, ins, timeout=None, group_id=0
+        )
+        self._common_assertions(ins)
 
-    def test_validate_task_res_missing_content(self) -> None:
-        """Test invalid TaskRes (missing content)."""
+    def test_fit_and_fail(self) -> None:
+        """Test negative case."""
         # Prepare
-        task_res = task_pb2.TaskRes(  # pylint: disable=E1101
-            task_id="554bd3c8-8474-4b93-a7db-c7bec1bf0012",
-            group_id="",
-            run_id=0,
-            task=task_pb2.Task(  # pylint: disable=E1101
-                error=error_pb2.Error(  # pylint: disable=E1101
-                    code=0,
-                    reason="Some reason",
-                )
-            ),
+        self.driver.push_messages.side_effect = self._get_push_messages(
+            None, error_reply=True
         )
+        parameters = flwr.common.ndarrays_to_parameters([np.ones((2, 2))])
+        ins = FitIns(parameters, {})
 
-        # Execute & assert
-        with self.assertRaises(ValueError):
-            validate_task_res(task_res=task_res)
-
+        # Execute and assert
+        self.assertRaises(Exception, self.client.fit, ins, timeout=None, group_id=0)
+        self._common_assertions(ins)
 
-class TestWithGrpcDriver(DriverClientProxyTestCase):
-    """Unit test using GrpcDriver."""
+    def test_evaluate_and_fail(self) -> None:
+        """Test negative case."""
+        # Prepare
+        self.driver.push_messages.side_effect = self._get_push_messages(
+            None, error_reply=True
+        )
+        parameters = Parameters(tensors=[b"random params%^&*F"], tensor_type="np")
+        ins = EvaluateIns(parameters, {})
 
-    __test__ = True
+        # Execute and assert
+        self.assertRaises(
+            Exception, self.client.evaluate, ins, timeout=None, group_id=0
+        )
+        self._common_assertions(ins)
+
+    def _create_message_dummy(  # pylint: disable=R0913
+        self,
+        content: RecordSet,
+        message_type: str,
+        dst_node_id: int,
+        group_id: str,
+        ttl: Optional[float] = None,
+    ) -> Message:
+        """Create a new message.
+
+        This is a method for the Mock object.
+        """
+        self.called_times += 1
+        ttl_ = 123456 if ttl is None else ttl
+        metadata = Metadata(
+            run_id=RUN_ID,
+            message_id="",  # Will be set by the server
+            src_node_id=0,
+            dst_node_id=dst_node_id,
+            reply_to_message="",
+            group_id=group_id,
+            ttl=ttl_,
+            message_type=message_type,
+        )
+        self.created_msg = Message(metadata=metadata, content=content)
+        if message_type == "evaluate":
+            print(compat.recordset_to_evaluateins(self.created_msg.content, True))
```

Comment:
Do we want to keep this?

Reply:
No. It was me testing why `tensor_type` is lost after serialization. Will remove it. Thanks.